Sources/TuistGenerator/Mappers/AutogeneratedProjectSchemeWorkspaceMapper.swift

```diff
@@ -66,7 +68,13 @@ public final class AutogeneratedProjectSchemeWorkspaceMapper: WorkspaceMapping {
                     .filter { $0.platform == platform }
                     .filter(\.product.testsBundle)
                     .map { TargetReference(projectPath: project.path, name: $0.name) }
-                    .map { TestableTarget(target: $0, parallelizable: true) }
+                    .map {
+                        TestableTarget(
+                            target: $0,
+                            parallelizable: testingOptions?.contains(.parallelizable) ?? true,
```

Comment:
As commented in #3764 I would suggest reverting parallelizable to `false`, the majority of tests targets are usually unit tests. Unit tests are usually faster when not run in parallel (Xcode does not need to clone the simulators which can take a while). Before changing, let's wait for @danyf90 and @fortmarek if they agree with the suggestion 😃

Reply:
I would love to read more about this behavior if you have a good resource. But knowing this, I agree default should be `false` if it will make majority of projects' tests run faster.

Reply:
If this is just a simulator issue, then default should remain `true` where simulator isn't being used.

Additionally, is it possible to determine whether the tests are unit tests?

Reply:
> If this is just a simulator issue, then default should remain true where simulator isn't being used.

This is a good point - I think this depends on whether the `.app` target is imported or not. In most modular apps I've seen, the majority of frameworks is independent from the app code, so `parallelizable` will make it faster. We should be able to make the right default based on the imports but I'd say this might be out of scope for this PR. I suggest keeping it to `false` as default to keep the previous behavior and only then implement the `parallelizable` heuristics. 

> Additionally, is it possible to determine whether the tests are unit tests?

This depends on the `Target.product`, so yes, it should be

Reply:
To make sure I understand, the expectation here is (using `parallelizable` as an example but applies for test order as well):

```
if parallelizable value set in config {
    - use config value for unit tests 
    - use config value for UI tests
} else {
    - use default of _false_ for unit tests 
    - use default of _true_ for UI tests
}
```

Reply:
@wattson12 let's just go for `false` as default for now, we can always give a better thought in a different PR :)

Reply:
Definitely think this should stay false by default. The Xcode default for this is false and moving things to true creates some sometimes unwanted side effects since it does change test ordering (which yes, teams shouldn't depend on, but that's the status quo) as well as questionable performance gains since for small test targets the overhead on parallel sims is more than just running quickly with one.

Reply:
This is done now: the default goes back to false and users can configure as needed